src/Quadrature.jl

```diff
@@ -399,23 +399,27 @@ ZygoteRules.@adjoint function __solvebp(prob,alg,sensealg,lb,ub,p,args...;kwargs
     function quadrature_adjoint(Δ)
         y = typeof(Δ) <: Array{<:Number,0} ? Δ[1] : Δ
         if isinplace(prob)
-            if lb isa Number && prob.batch == 0
-                dx = Float64[lb]
-            elseif lb isa Number
-                dx = zeros(length(lb),prob.batch)
-            elseif prob.batch == 0
-                dx = zeros(length(lb))
-            else
-                dx = zeros(length(lb),prob.batch)
-            end
-            _f = (x) -> (prob.f(dx,x,p); dx)
+            # if lb isa Number && prob.batch == 0
+            #     dx = Float64[lb]
+            # elseif lb isa Number
+            #     dx = zeros(length(lb),prob.batch)
+            # elseif prob.batch == 0
+            #     dx = zeros(length(lb))
+            # else
+            #     # dx = zeros(length(lb),prob.batch)
+            #     dx = zeros(prob.nout, prob.batch)
+            # end
+            # _f = (x) -> (prob.f(dx,x,p); dx)
+
+            # @show size(dx)
 
             if sensealg.vjp isa ZygoteVJP
-                dfdp = function (dx,x,p)
+                # dfdp = function (dx,x,p)
+                dfdp = function (x,p)
                     _,back = Zygote.pullback(p) do p
-                        dx = Zygote.Buffer(x)
-                        prob.f(dx,x,p)
-                        copy(dx)
+                        _dx = Zygote.Buffer(x, prob.nout, size(x,2))
```

Comment:
For some of the quadrature methods `prob.batch` isn't adhered to. It looks like it serves more as a max batch number. Some methods "grow" the batch size. So, need to set the solution size accordingly.

Reply:
Further more, Quadrature.jl runs a test calling `prob.f([lb ub], p)` to test if the solution of the integrand is a `Vector`. So if `batch>0`, the code will always try two point batch first.